### PR TITLE
feat(secrets): Inject secrets omitter to runner registry

### PR DESCRIPTION
**Pull request by Eliran-Turgeman**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description
* Inject the secrets omitter class name into the runner registry
* Remove s3-related implementation in secrets runner as checkov only accesses the reports given as an argument

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/common/runners/runner_registry.py [hunk 1]
```diff
@@ -54,7 +53,8 @@
 
 
 class RunnerRegistry:
-    def __init__(self, banner: str, runner_filter: RunnerFilter, *runners: _BaseRunner) -> None:
+    def __init__(self, banner: str, runner_filter: RunnerFilter, *runners: _BaseRunner,
+                 secrets_omitter_class_name: str | None = None) -> None:
```
Comment (nimrodkor) [inline checkov/common/runners/runner_registry.py:57]:
```suggestion
    def __init__(self, banner: str, runner_filter: RunnerFilter, *runners: _BaseRunner,
                 secrets_omitter_class: Type[SecretsOmitter] | None = SecretsOmitter) -> None:
```

File: checkov/common/runners/runner_registry.py [hunk 2]
```diff
@@ -101,8 +102,11 @@ def _parallel_run(runner: _BaseRunner) -> Report | list[Report]:
             reports = parallel_runner.run_function(func=_parallel_run, items=self.runners, group_size=1)
 
         merged_reports = self._merge_reports(reports)
-        if bc_integration.bc_api_key:
-            SecretsOmitter(merged_reports).omit()
+
+        if self.secrets_omitter_class_name and bc_integration.bc_api_key:
+            secrets_omitter_class_obj = globals().get(self.secrets_omitter_class_name)
+            if secrets_omitter_class_obj:
+                secrets_omitter_class_obj(merged_reports).omit()
```
Comment (nimrodkor) [inline checkov/common/runners/runner_registry.py:109]: @Eliran-Turgeman if you pass in the class, then:
```suggestion
        if self.secrets_omitter_class_name and bc_integration.bc_api_key:
            self.secrets_omitter_class_name(merged_reports).omit()
```

File: checkov/main.py [hunk 3]
```diff
@@ -202,6 +203,7 @@ def run(banner: str = checkov_banner, argv: List[str] = sys.argv[1:]) -> Optiona
     else:
         runner_registry = RunnerRegistry(banner, runner_filter, *DEFAULT_RUNNERS)
 
+    runner_registry.secrets_omitter_class_name = SecretsOmitter.__name__
```
Comment (nimrodkor) [inline checkov/main.py:206]:
```suggestion
        runner_registry = RunnerRegistry(banner, runner_filter, *DEFAULT_RUNNERS, secrets_omitter_class=SecretsOmitter)
```

File: checkov/common/runners/runner_registry.py [hunk 4]
```diff
@@ -64,6 +64,7 @@ def __init__(self, banner: str, runner_filter: RunnerFilter, *runners: _BaseRunn
         self.filter_runner_framework()
         self.tool = tool_name
         self._check_type_to_report_map: dict[str, Report] = {}  # used for finding reports with the same check type
+        self.secrets_omitter_class_name = secrets_omitter_class_name
```
Comment (nimrodkor) [inline checkov/common/runners/runner_registry.py:67]:
```suggestion
        self.secrets_omitter_class = secrets_omitter_class
```

**Review by nimrodkor — APPROVED** [review]

File: checkov/common/runners/runner_registry.py [hunk 5]
```diff
@@ -54,7 +54,8 @@
 
 
 class RunnerRegistry:
-    def __init__(self, banner: str, runner_filter: RunnerFilter, *runners: _BaseRunner) -> None:
+    def __init__(self, banner: str, runner_filter: RunnerFilter, *runners: _BaseRunner,
+                 secrets_omitter_class: Type[SecretsOmitter] | None = SecretsOmitter) -> None:
```
Comment (arielkru) [inline checkov/common/runners/runner_registry.py:58]: When is `secrets_omitter_class = None`?

Comment (Eliran-Turgeman) [reply checkov/common/runners/runner_registry.py:58]: Never. good catch!

**Review by arielkru — APPROVED** [review]

**Pull request merged**